File: src/com/aopphp/go/pointcut/PointFilter.java
package com.aopphp.go.pointcut;

import com.jetbrains.php.lang.psi.elements.PhpNamedElement;

import java.io.Serializable;
import java.util.Set;

/**
 * Filter that restricts matching of a pointcut or introduction to a given set of reflection points.
 */
public interface PointFilter extends Serializable
{
    boolean matches(PhpNamedElement element);

    Set<KindFilter> getKind();
}

File: src/main/kotlin/com/aopphp/go/pointcut/PointFilter.kt
package com.aopphp.go.pointcut

import com.jetbrains.php.lang.psi.elements.PhpNamedElement
import java.io.Serializable

interface PointFilter : Serializable {
    fun matches(element: PhpNamedElement): Boolean
    fun getKind(): Set<KindFilter>
}
